scripts/chgres_cube.py

```diff
@@ -0,0 +1,288 @@
+
+
+def _parse_var_defns(file):
+    var_dict = {}
+    with open(file, "r", encoding="utf-8") as f:
+        lines = f.readlines()
+        for line in lines:
+            line = line.strip()
+            if "=" in line:
+                key, value = line.split("=", 1)
+                key = key.strip()
+                value = value.strip()
+
+                if value.startswith("(") and value.endswith(")"):
+                    items = re.findall(r"\((.*?)\)", value)
+                    if items:
+                        value = [item.strip() for item in items[0].split()]
+                        var_dict[key] = value
+    return var_dict
+
+
+def _walk_key_path(config, key_path):
+    """
+    Navigate to the sub-config at the end of the path of given keys.
+    """
+    keys = []
+    pathstr = "<unknown>"
+    for key in key_path:
+        keys.append(key)
+        pathstr = " -> ".join(keys)
+        try:
+            subconfig = config[key]
+        except KeyError:
+            logging.error(f"Bad config path: {pathstr}")
+            raise
+        if not isinstance(subconfig, dict):
+            logging.error(f"Value at {pathstr} must be a dictionary")
+            sys.exit(1)
+        config = subconfig
+    return config
+
+
+def parse_args(argv):
+    """
+    Parse arguments for the script.
+    """
+    parser = ArgumentParser(
+        description="Script that runs chgres_cube via uwtools API",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        metavar="PATH",
+        required=True,
+        help="Path to experiment config file.",
+        type=Path,
+    )
+    parser.add_argument(
+        "--cycle",
+        help="The cycle in ISO8601 format (e.g. 2024-07-15T18)",
+        required=True,
+        type=dt.datetime.fromisoformat,
+    )
+    parser.add_argument(
+        "--key-path",
+        help="Dot-separated path of keys leading through the config to the driver's YAML block",
+        metavar="KEY[.KEY...]",
+        required=True,
+        type=lambda s: s.split("."),
+    )
+    parser.add_argument(
+        "--member",
+        default="000",
+        help="The 3-digit ensemble member number.",
+    )
+    return parser.parse_args(argv)
+
+
+# pylint: disable=too-many-locals, too-many-statements, too-many-branches
+def run_chgres_cube(config_file, cycle, key_path, member):
+    """
+    Setup and run the chgres_cube Driver.
+    """
+
+    # dereference expressions during driver initialization
+    expt_config = get_yaml_config(config_file)
+    CRES = expt_config["workflow"]["CRES"]
+    os.environ["CRES"] = CRES
+    os.environ["MEMBER"] = member
+
+    # set universal variables
+    cyc = str(expt_config["workflow"]["DATE_FIRST_CYCL"])[8:10]
+    dot_ensmem = (
+        f".mem{member}"
+        if (
+            expt_config["user"]["RUN_ENVIR"] == "nco"
+            and expt_config["global"]["DO_ENSEMBLE"]
+            and member
+        )
+        else ""
+    )
```

Comment:
I ended up keeping the `nco` part as it was in the full original condition, but it can be nicer. I don't see any reason your oneliner wouldn't work, thank you!